### PR TITLE
Improve docs of links/connection/lstm

### PR DESCRIPTION
**Pull request by keisuke-umezawa**
This is PR for improving docs of functions/links. Related issue: #2182

Modified functions/links:

- links/connection/lstm.py

### CODE REVIEW

**Comment by mitmul** [comment]
@keisuke-umezawa Could you rebase it to the current master?

File: chainer/links/connection/lstm.py [hunk 1]
```diff
@@ -199,6 +199,35 @@ class LSTM(LSTMBase):
         c (~chainer.Variable): Cell states of LSTM units.
         h (~chainer.Variable): Output at the previous time step.
 
+    .. admonition:: Example
+
+        There are several ways to make a LSTM link.
+
+        Let a two-dimensional input array :math:`x`, a cell state array
+        :math:`h`, and the output array of the previous step :math:`h` be:
```
Comment (keisuke-umezawa) [inline chainer/links/connection/lstm.py]: fixed it.

**Comment by keisuke-umezawa** [comment]
@mitmul I rebased and followed review.

**Comment by mitmul** [comment]
Jenkins, test this please

**Comment by mitmul** [comment]
LGTM

**Pull request merged**